src/Razor/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -517,6 +517,238 @@ public class ComponentWithNullableRenderFragment : ComponentBase
+            Assert.Equal(RazorDiagnosticSeverity.Warning, diagnostics.Severity);
+            Assert.Equal("RZ2012", diagnostics.Id);
+        }
+
+        [Fact]
+        public void Component_WithEditorRequiredChildContent_ValueSpecified_WithoutName()
+        {
+            AdditionalSyntaxTrees.Add(Parse(@"
+using System;
+using Microsoft.AspNetCore.Components;
+namespace Test
+{
+    public class ComponentWithEditorRequiredChildContent : ComponentBase
+    {
+        [Parameter]
+        [EditorRequired]
+        public RenderFragment ChildContent { get; set; }
+    }
+}
+"));
+            var generated = CompileToCSharp(@"
+<ComponentWithEditorRequiredChildContent>
+    <h1>Hello World</h1>
+</ComponentWithEditorRequiredChildContent>
+
+");
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+
+            Assert.Empty(generated.Diagnostics);
+        }
+
+        [Fact]
+        public void Component_WithEditorRequiredChildContent_ValueSpecified()
+        {
+            AdditionalSyntaxTrees.Add(Parse(@"
+using System;
+using Microsoft.AspNetCore.Components;
+namespace Test
+{
+    public class ComponentWithEditorRequiredChildContent : ComponentBase
+    {
+        [Parameter]
+        [EditorRequired]
+        public RenderFragment ChildContent { get; set; }
+    }
+}
+"));
+            var generated = CompileToCSharp(@"
+<ComponentWithEditorRequiredChildContent>
+    <ChildContent>
+        <h1>Hello World</h1>
+    </ChildContent>
+</ComponentWithEditorRequiredChildContent>
+
+");
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated);
+
+            Assert.Empty(generated.Diagnostics);
+        }
+
+        [Fact]
+        public void Component_WithEditorRequiredNamedChildContent_NoValueSpecified()
+        {
+            AdditionalSyntaxTrees.Add(Parse(@"
+using System;
+using Microsoft.AspNetCore.Components;
+namespace Test
+{
+    public class ComponentWithEditorRequiredChildContent : ComponentBase
+    {
+        [Parameter]
+        [EditorRequired]
+        public RenderFragment Found { get; set; }
+
+        [Parameter]
+        public RenderFragment NotFound { get; set; }
+    }
+}
+"));
+            var generated = CompileToCSharp(@"
+<ComponentWithEditorRequiredChildContent>
+</ComponentWithEditorRequiredChildContent>
+
+");
+            AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
+            AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
+            CompileToAssembly(generated, throwOnFailure: false);
+
+            var diagnostics = Assert.Single(generated.Diagnostics);
+            Assert.Equal(RazorDiagnosticSeverity.Warning, diagnostics.Severity);
+            Assert.Equal("RZ2012", diagnostics.Id);
```

Comment:
This is a bit weird. It looks like there *is* child content:

```
<ComponentWithEditorRequiredChildContent>
</ComponentWithEditorRequiredChildContent>
```

Syntactically it looks like the child content is a "newline" character. What makes this illegal? Maybe you actually want a single newline character as child content. Does the Razor compiler special-case this into "no child content parameter"?

Overall since `[EditorRequired]` isn't default, I wouldn't expect most developers to put it on a `ChildContent` parameter. I suppose if they do put it on, then they are really determined to get some child content so perhaps "whitespace" should count as violating their rule.

**Summary**: it's not obvious what the right behavior should be, but I guess I'm fine with `[EditorRequired]` erring on the side of strictness since if the developer wants not to be strict, they just wouldn't use it in the first place.

What do you think?

Reply:
In this particular case, the child contents are named (it's `Found` and `NotFound` like the `Router` component), so this particular error makes sense. However, I wasn't sure what the behavior looked like when specifying text as a child content:

For a component with a parameter like so, here's the behavior:

```razor
[EditorRequired][Parameter] public RenderFragment ChildContent { get; set; }
```

* ✔️ Having non-whitespace text does not produce a warning eg. `<MyComponent>Hello world</MyComponent>`
* ✔️ Having any markup or component tag does not produce a warning e.g. `<MyComponent><Counter /></MyComponent>`
* ❌ One or more whitespace is treated as missing ChildContent which produces a warning: `<MyComponent></MyComponent>`
* ✔️ You can specify an explicit tag if you do want whitespace to be treated as valid content `<MyComponent><ChildContent> </ChildContent></MyComponent>`

I feel bullet no 3 should help guide the most common use case of the editor completing the tag for the user while 4 covers the case if you need to workaround any undue compiler restrictions.